native/src/main/java/io/ballerina/stdlib/kafka/utils/KafkaUtils.java

```diff
@@ -570,7 +571,7 @@ public static ArrayList<TopicPartition> getTopicPartitionList(BArray partitions,
         return partitionList;
     }
 
-    public static List<String> getStringListFromStringBArray(BArray stringArray) {
+    public static ArrayList<String> getStringListFromStringBArray(BArray stringArray) {
```

Comment:
Code smells son. This is better left as it was originally with it returning the type `List` instead of `ArrayList` there are reasons to why we do that and you should read about that. :D https://coderanch.com/t/408971/java/return-type-Interface-Class

Reply:
I narrowed the types because this resulted in a class cast exception for other types of Lists. But apparently the issue is not here. It is due to too narrow cast done in https://github.com/ballerina-platform/module-ballerinax-kafka/blob/master/native/src/main/java/io/ballerina/stdlib/kafka/impl/KafkaRecordConsumer.java#L75

Fixed this in